src/app_service/service/metrics/service.nim

```diff
@@ -0,0 +1,52 @@
+import json, chronicles, times
+include ../../common/json_utils
+
+import backend/response_type
+import status_go
+import constants
+import ./dto
+
+logScope:
+  topics = "metrics"
+
+proc toggleCentralizedMetrics*(enabled: bool) =
+  try:
+    let payload = %* {"enabled": enabled}
+    let response = status_go.toggleCentralizedMetrics($payload)
+    let jsonObj = response.parseJson
+    if jsonObj.hasKey("error"):
+      error "toggleCentralizedMetrics", errorMsg=jsonObj["error"].getStr
+  except Exception:
+    discard
+
+# for testing, needs to be discussed
+proc addCentralizedMetric*() =
+  try:
+    var metric = CentralizedMetricDto()
+    metric.userId = "123456"
+    metric.eventName = "desktop-event"
+    #metric.eventValue = """{"action": "section-changed"}"""
+    metric.timestamp = now().toTime().toUnix()
+    metric.platform = hostOS
+    metric.appVersion = APP_VERSION
+
+    let payload = %* {"metric": metric.toJsonNode}
+    echo "!!! Add metric: ", payload
+    let response = status_go.addCentralizedMetric($payload)
```

Comment:
I think maybe this should be done async since it's a write operation and we don't want to disrupt the user

Reply:
I will do it in one of next prs. Besides parameters of this function will change.
I changed the service to QObject in order to prepare for async calls.